jwk/jwk.go

```diff
@@ -744,3 +744,13 @@ func Equal(k1, k2 Key) bool {
 
 	return bytes.Equal(tp1, tp2)
 }
+
+// IsPrivate tells whether the supplied key is a private or public key.  An
+// error is raised if the supplied key is not an asymmetric key.
+func IsPrivate(k Key) (bool, error) {
```

Comment:
```suggestion
func IsPrivateKey(k Key) (bool, error) {
```

I think that this feels more natural:

```go
key.IsPrivate()
jwk.IsPrivateKey(key)
```